src/main/java/com/tradeshift/resources/HelloWorldResource.java

```diff
@@ -0,0 +1,52 @@
+package com.tradeshift.resources;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+
+import com.tradeshift.model.Content;
+import com.tradeshift.model.Message;
+import com.tradeshift.model.RecentModel;
+import com.tradeshift.service.HelloWorldService;
+
+
+@Path("messages")
+@Controller
+public class HelloWorldResource {
+
+    private HelloWorldService helloWorldService;
+
+    @Autowired
+    public HelloWorldResource(HelloWorldService helloWorld) {
+        this.helloWorldService = helloWorld;
+    }
+
+    @POST
+    @Path("/names/{name}")
+    @Produces("application/json")
+    @Consumes({"text/plain,text/html"})
+    public Message insertNameToDB(@PathParam("name") String name) {
+        String content;
+        try {
+            content = helloWorldService.getHelloWorldMessage(name);
+            helloWorldService.insert(name);
+        } catch (NullPointerException e) {
+            content = "Name can not be empty";
```

Comment:
This is a bad request and should be handled as such. I know that it isn't in the requirements, but it's REST conventions. It's also controversial to catch NPE's. It's better then to check for null up front, for example ```java
if (name == null) {
  throw new BadRequestException();
} else {
  // do stuff 
}```